libraries/AP_Notify/Buzzer.cpp

```diff
@@ -68,15 +68,57 @@ void Buzzer::update()
 
 void Buzzer::update_pattern_to_play()
 {
+    if (!_flags.initialize_done) {
+        // play a quick beep at startup
+        if (!_flags.startup_beep) {
+            _flags.startup_beep = true;
+            play_pattern(SINGLE_BUZZ);
+            return;
+        }
+        // play beeps when system initialization complete
+        if (!_flags.initialize_started) {
+            if (AP_Notify::flags.initialising) {
+                _flags.initialize_started = true;
+            }
+        } else if (!AP_Notify::flags.initialising) {
+            _flags.initialize_done = true;
+            play_pattern(INITIALIZE_BUZZ);
+            return;
+        }
+    }
+
+    // check if radio failsafe status has changed
+    if (_flags.failsafe_radio != AP_Notify::flags.failsafe_radio) {
+        _flags.failsafe_radio = AP_Notify::flags.failsafe_radio;
+        // play beeps to indicate radio status (if initialization completed)
+        if (_flags.initialize_done) {
```

Comment:
Do we want to play*any* tones before initialisation is done?  If not, this can be moved out/up.